### PR TITLE
SmartStore tests fixes

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/ui/SmartStoreInspectorActivity.java [hunk 1]
```diff
@@ -125,8 +125,8 @@ protected void onResume() {
 
 	private void readExtras() {
 		Bundle bundle = getIntent().getExtras();
-		isGlobal = bundle.getBoolean(IS_GLOBAL_STORE, false);
-		dbName = bundle.getString(DB_NAME, DBOpenHelper.DEFAULT_DB_NAME);
+		isGlobal = bundle == null ? false : bundle.getBoolean(IS_GLOBAL_STORE, false);
+		dbName = bundle == null ? DBOpenHelper.DEFAULT_DB_NAME : bundle.getString(DB_NAME, DBOpenHelper.DEFAULT_DB_NAME);
```
Comment (wmathurin) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/ui/SmartStoreInspectorActivity.java]: SmartStoreInspectorTest was throwing a NPE here

File: libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java [hunk 2]
```diff
@@ -600,7 +600,7 @@ public void dropAllSoups() {
 	    	List<String> soupNames = new ArrayList<String>();
 	        Cursor cursor = null;
 			try {
-				cursor = DBHelper.getInstance(db).query(db, SOUP_NAMES_TABLE, new String[]{SOUP_NAME_COL}, null, null, null);
+				cursor = DBHelper.getInstance(db).query(db, SOUP_NAMES_TABLE, new String[]{SOUP_NAME_COL}, SOUP_NAME_COL, null, null);
```
Comment (wmathurin) [inline libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java]: SmartStoreInspectorTest expected results in a certain order

File: libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreFullTextSearchSpeedTest.java [hunk 3]
```diff
@@ -148,7 +148,7 @@ private double setupData(Type textFieldType, int rowsPerAnimal, int matchingRows
     private double queryData(Type textFieldType, int rowsPerAnimal, int matchingRowsPerAnimal) throws JSONException {
         long totalQueryTime = 0;
         for (String animal : ANIMALS) {
-            String prefix = String.format("%07d", (int) Math.random()*matchingRowsPerAnimal);
+            String prefix = String.format("%07d", (int) (Math.random()*(rowsPerAnimal/matchingRowsPerAnimal)));
```
Comment (wmathurin) [inline libs/test/SmartStoreTest/src/com/salesforce/androidsdk/store/SmartStoreFullTextSearchSpeedTest.java]: We were always using 0000000 prefix

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**